google/cloud/aiplatform/models.py

```diff
@@ -15,9 +15,213 @@
+                container's environment. If a variable cannot be resolved, the reference
+                in the input string will be unchanged. The $(VAR_NAME) syntax can be
+                escaped with a double $$, ie: $$(VAR_NAME). Escaped references will
+                never be expanded, regardless of whether the variable exists or not.
+            serving_container_environment_variables: Optional[Dict[str, str]]=None,
+                The environment variables that are to be present in the container.
+                Should be a dictionary where keys are environment variable names
+                and values are environment variable values for those names.
+            serving_container_ports: Optional[Sequence[int]]=None,
+                Declaration of ports that are exposed by the container. This field is
+                primarily informational, it gives AI Platform information about the
+                network connections the container uses. Listing or not a port here has
+                no impact on whether the port is actually exposed, any port listening on
+                the default "0.0.0.0" address inside a container will be accessible from
+                the network.
+            project: Optional[str]=None,
+                Project to upload this model to. Overrides project set in
+                aiplatform.init.
+            location: Optional[str]=None,
+                Location to upload this model to. Overrides location set in
+                aiplatform.init.
+            credentials: Optional[auth_credentials.Credentials]=None,
+                Custom credentials to use to upload this model. Overrides credentials
+                set in aiplatform.init.
+        Returns:
+            model: Instantiated representation of the uplaoded model resource.
+        """
+
+        api_client = cls._instantiate_client(location, credentials)
+        env = None
+        ports = None
+
+        if serving_container_environment_variables:
+            env = [
+                env_var.EnvVar(name=str(key), value=str(value))
+                for key, value in serving_container_environment_variables.items()
+            ]
+        if serving_container_ports:
+            ports = [
+                gca_model.Port(container_port=port) for port in serving_container_ports
+            ]
+
+        container_spec = gca_model.ModelContainerSpec(
+            image_uri=serving_container_image_uri,
+            command=serving_container_command,
+            args=serving_container_args,
+            env=env,
+            ports=ports,
+            predict_route=serving_container_predict_route,
+            health_route=serving_container_health_route,
+        )
+
+        managed_model = gca_model.Model(
+            display_name=display_name,
+            description=description,
+            artifact_uri=artifact_uri,
+            container_spec=container_spec,
+        )
+
+        lro = api_client.upload_model(
```

Comment:
When the `LRO` class is in place, we will use an instance of that class instead?

Reply:
@sirtorry FYI.

Reply:
Yes.